include/util/deallocating_vector.hpp

```diff
@@ -258,8 +258,12 @@ template <typename ElementT> class DeallocatingVector
     DeallocatingVector(DeallocatingVector &other) = delete;
     DeallocatingVector &operator=(DeallocatingVector &other) = delete;
     // moving is fine
-    DeallocatingVector(DeallocatingVector &&other) = default;
-    DeallocatingVector &operator=(DeallocatingVector &&other) = default;
+    DeallocatingVector(DeallocatingVector &&other) { swap(other); }
+    DeallocatingVector &operator=(DeallocatingVector &&other)
```

Comment:
I know where this came from, still I am a bit confused on why this change was necessary. Was it just unexplainable behaviour, or did you find some reason for the problem?